### PR TITLE
[darwin]: MacOS compatible build

### PR DESCRIPTION
**Pull request by jkuri**
This PR adds compatibility to build static FFmpeg builds right off running `./build.sh` script and in same time doesn't affect Linux builds. 

I also updated FFmpeg version to `4.0` and make build a bit more static with adding additional libraries to be built statically.

### CODE REVIEW

File: fetchurl [hunk 1]
```diff
@@ -104,10 +104,10 @@ if [ "$UNPACK" -ne 0 ]; then
   target_dir=`expand_path "$EXTRACT_DIR"`
   mkdir -p "$target_dir"
   sh cd "$target_dir"
- 
-  [ "$REPLACE" -ne 1 ] && tarargs="--skip-old-files" || tarargs=""
+
+  [ "$REPLACE" -ne 1 ] && tarargs=""
```
Comment (zimbatm) [inline fetchurl]: isn't this breaking the REPLACE feature?

**Comment by jkuri** [comment]
you are correct, I need to check OS before that, —skip-old-files is not
supported on darwins tar.

**Comment by zimbatm** [comment]
okay, let me know once it's done and I will merge the PR

**Comment by jkuri** [comment]
@zimbatm updated. please check again if it looks good to you.

**Pull request merged**